compiler/rustc_parse/src/lexer/tokentrees.rs

```diff
@@ -238,13 +235,29 @@ impl<'psess, 'src> Lexer<'psess, 'src> {
         this_spacing
     }
 
-    fn close_delim_err(&mut self, delim: Delimiter) -> Diag<'psess> {
+    fn close_delim_err(&mut self, close_delim: Delimiter) -> Diag<'psess> {
         // An unexpected closing delimiter (i.e., there is no matching opening delimiter).
         let token_str = token_to_string(&self.token);
         let msg = format!("unexpected closing delimiter: `{token_str}`");
         let mut err = self.dcx().struct_span_err(self.token.span, msg);
 
-        report_suspicious_mismatch_block(&mut err, &self.diag_info, self.psess.source_map(), delim);
+        if let Some((open_delim, open_delim_span)) = self.diag_info.open_delimiters.last().cloned()
+        {
+            if open_delim == close_delim {
+                err.span_label(
+                    open_delim_span,
+                    format!("the mismatchd closing `{}` may be matched here", token_str),
```

Comment:
This message has a typo, and I'd like to tweak the wording here a bit, I just don't have a good idea for what the ideal text would be in context.